src/component/panels/extra/utilities/ZoneUtilities.js

```diff
@@ -51,4 +59,89 @@ const unlink = (zone, isOnZoneLevel, signalIndex, axis) => {
   return zoneObject;
 };
 
-export { getDiaIDs, getPubIntegral, resetDiaIDs, unlink };
+const unlinkInAssignmentData = (
+  assignmentData,
+  zone,
+  isOnZoneLevel,
+  signalIndex,
+  axis,
+) => {
+  if (isOnZoneLevel !== undefined && axis !== undefined) {
+    if (isOnZoneLevel === true) {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: zone.id,
+          axis: axis,
+        },
+      });
+    } else if (signalIndex !== undefined) {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${signalIndex}`,
+          axis: axis,
+        },
+      });
+    }
+  } else if (axis !== undefined) {
+    assignmentData.dispatch({
+      type: 'REMOVE_ALL',
+      payload: {
+        id: zone.id,
+        axis: axis,
+      },
+    });
+    zone.signal.forEach((_signal, i) => {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${i}`,
+          axis: axis,
+        },
+      });
+    });
+  } else {
+    assignmentData.dispatch({
+      type: 'REMOVE_ALL',
+      payload: {
+        id: zone.id,
+        axis: 'x',
+      },
+    });
+    zone.signal.forEach((_signal, i) => {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${i}`,
+          axis: 'x',
+        },
+      });
+    });
+    assignmentData.dispatch({
+      type: 'REMOVE_ALL',
+      payload: {
+        id: zone.id,
+        axis: 'y',
+      },
+    });
+    zone.signal.forEach((_signal, i) => {
+      assignmentData.dispatch({
+        type: 'REMOVE_ALL',
+        payload: {
+          id: `${zone.id}${HighlightSignalConcatenation}${i}`,
+          axis: 'y',
+        },
+      });
```

Comment:
Do not use Dispatch inside any kind of for loop, all the logic must be don inside the reducers and not inside the component, this better for separation and you must reflect this to all your work that you don.